uk.ac.diamond.ispyb.scanning/src/uk/ac/diamond/ispyb/scanning/ExperimentCommunicationService.java

```diff
@@ -107,4 +114,40 @@ public static void setIspybXpdfFactoryService(IspybXpdfFactoryService ispybXpdfF
 		}
 	}
 
+	@Override
+	public SampleInformation getSampleInformation(String proposalCode, long proposalNumber, long sampleId) {
+		return getSampleInformation(proposalCode, proposalNumber, new long[]{sampleId}).values().iterator().next();
+	}
+
+	@Override
+	public Map<Long, SampleInformation> getSampleInformation(String proposalCode, long proposalNumber, long... sampleIds) {
+		
+		if (sampleIds==null || sampleIds.length<1) throw new IllegalArgumentException("At least one sampleId must be provided");
+		
+		List<Long>   requiredIds = Arrays.stream(sampleIds).boxed().collect(Collectors.toList());
+		List<Sample> samples     = getSamples(proposalCode, proposalNumber);
+		List<Sample> rsamples    = samples.stream().filter(sample->requiredIds.contains(sample.getSampleId())).collect(Collectors.toList());

```

Comment:
Don't think this one can be shorter as in the filter you are calling 2 instance methods requiredIds.contains and sample.getSampleId